qemu/tests/kvm_stat.py

```diff
@@ -0,0 +1,82 @@
+from virttest import utils_misc
+from virttest import utils_package
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1.install kernel-tools package in host
+    2.modify kvm_stat.service file and start kvm_stat.service
+    3.run VM
+    4.use "logrotate" to separate the log file
+    5.Check the generated log file
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    interval = float(params.get("interval_time", "10"))
+
+    def tweak_file(filename, old_str, new_str):
+        # replace the strings in the file
+        file_data = ""
+        with open(filename, "r") as f:
+            for line in f:
+                if old_str in line:
+                    line = line.replace(old_str, new_str, 1)
+                file_data += line
+        with open(filename, "w") as f:
+            f.write(file_data)
+
+    def separate_file(file_path):
+        num = 1
+        file_size = os.path.getsize(file_path)
+        while num < 6:
+            if file_size > 10240:
+                process.system('logrotate -v /etc/logrotate.d/kvm_stat')
+            num += 1
+
+    def check_log():
+        check_file = params.get("check_file")
+        check_file_cmd = subprocess.Popen(check_file, shell=True, stdout=subprocess.PIPE)
+        out = check_file_cmd.stdout.readlines()
+        for line in out:
+            result = line.strip().decode()
+            if 'kvm_stat.csv.5.gz' in result:
+                logging.info("Test complete")
+
+    depends_pkgs = params.objects("depends_pkgs")
+    logging.info("Install packages: %s in host", depends_pkgs)
+    if not utils_package.package_install(depends_pkgs):
+        test.cancel("Install %s packages failed", depends_pkgs)
+    kvm_stat_service_path = params.get("kvm_stat_service_path")
+    kvm_stat_interval = params.get("kvm_stat_interval")
+    logging.info("Adjust the parameter '-s' is %s in %s", kvm_stat_interval, kvm_stat_service_path)
+    if os.path.exists(kvm_stat_service_path):
+        tweak_file(kvm_stat_service_path, "10", kvm_stat_interval)
+    else:
+        test.error("%s file does not exist", kvm_stat_service_path)
+    logging.info("Start kvm_stat.service")
+    kvm_stat_start_cmd = params.get("kvm_stat_start_cmd")
+    start_service = process.system(kvm_stat_start_cmd, timeout=interval, shell=True)
+    if start_service != 0:
+        test.error("Failed to start the kvm_stat.server")
+    else:
+        logging.info("Successfully to start the kvm_stat.service")
+    params["start_vm"] = 'yes'
+    env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+    vm = env.get_vm(params["main_vm"])
+    logging.info("Start logrotate command")
+    file_size = params.get("file_size")
+    tweak_file("/etc/logrotate.d/kvm_stat", "10M", file_size)
+    log_path = params.get("log_path")
+    utils_misc.wait_for(lambda: separate_file(log_path), 40)
```

Comment:
From manual case, there is one step to check log file number do not more than `rotate` in `/etc/logrotate.d/kvm_stat`.
If you are using `while num < 6:` in `separate_file`, it is not make sense.